plugins/modules/zendesk_ticket.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Luis Valle (levalle232@gmail.com)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: zendesk_ticket
+short_description: Manages tickets in Zendesk
+description:
+  - This module allows you to create and delete tickets in Zendesk.
+  - Authentication is handled by the ZENDESK_API class.
+author: "Luis Valle (@elchico2007)"
+version_added: 9.2.0
+extends_documentation_fragment:
+    - community.general.attributes
+attributes:
+  check_mode:
+    support: none
+  diff_mode:
+    support: none
+options:
+  host:
+    type: str
+    description: The URL of the Zendesk instance.
+    required: true
+  username:
+    type: str
+    description: The Zendesk account username.
+    required: true
+    aliases: ['user']
+  password:
+    type: str
+    description: The Zendesk account password.
+    required: false
+    aliases: ['pass']
+  token:
+    type: str
+    description: The API token for authentication.
+    required: false
+  body:
+    type: str
+    description: The body of the ticket.
+    default: ''
+  priority:
+    type: str
+    description: The priority of the ticket.
+    choices: ['urgent', 'high', 'normal', 'low']
+    default: normal
+  status:
+    type: str
+    description: The status of the ticket.
+    choices: ['new', 'closed', 'resolved']
```

Comment:
What is the difference between `closed` and `resolved` in Zendesk?

Reply:
With closed and no body is provided, the ticket will just be closed with no resolution. With resolved and a body, it will show the resolution of the ticket.

Reply:
Hi, reopening for a follow-up question. Would not be simpler to have one single `status` and, depending on whether a `body` is provided or not, for it to fall into one situation or the other?

Reply:
Hola, managed to do this and it is in one of the recent commits.